classes/Tools.php

```diff
@@ -88,29 +93,33 @@ public static function resetRequest()
      *
      * @return bool|string Password
      */
-    public static function passwdGen($length = 8, $flag = 'ALPHANUMERIC')
+    public static function passwdGen($length = 8, $flag = null)
     {
         $length = (int) $length;
 
         if ($length <= 0) {
             return false;
         }
 
+        if (null === $flag) {
+            $flag = static::PASSWORDGEN_FLAG_ALPHANUMERIC;
+        }
+
```

Comment:
```suggestion
```

Can be removed, it will go in default statement, otherwise the default has no aim and must raise an exception

Reply:
Yep!